internal/rbd/nodeserver.go

```diff
@@ -709,8 +709,12 @@ func (ns *NodeServer) NodePublishVolume(
 	volID := req.GetVolumeId()
 	stagingPath += "/" + volID
 
-	// Considering kubelet make sure the stage and publish operations
-	// are serialized, we dont need any extra locking in nodePublish
+	if acquired := ns.VolumeLocks.TryAcquire(targetPath); !acquired {
+		log.ErrorLog(ctx, util.TargetPathOperationAlreadyExistsFmt, targetPath)
+
+		return nil, status.Errorf(codes.Aborted, util.TargetPathOperationAlreadyExistsFmt, targetPath)
+	}
+	defer ns.VolumeLocks.Release(targetPath)
 
 	// Check if that target path exists properly
 	notMnt, err := ns.createTargetMountPath(ctx, targetPath, isBlock)
```

Comment:
If I see it correctly, `createTargetMountPath` still uses the (possibly unreliable?) `IsLikelyNotMountPoint` under the hood; should this be adjusted to the same `IsMountPoint` that the cephfs implementation uses? Or can this remain as-is, as discussed in https://github.com/ceph/ceph-csi/issues/4966#issuecomment-2488145516?

Reply:
Yes thats correct, it checks the sources, The main problem is with lock and the os.RemoveAll, if os.Remove fails we will get another RPC for unpublish which will trigger `umount` if its actually a mount.  For now lets keep it simple with what we  have today.